spark3/src/test/java/org/apache/iceberg/actions/TestCreateActions.java

```diff
@@ -0,0 +1,272 @@
+    String source = uniqueName("test_properties_table");
+    Map<String, String> props = Maps.newHashMap();
+    props.put("city", "New Orleans");
+    props.put("note", "Jazz");
+    testCreate(source,
+        dest,
+        "CREATE TABLE %s using parquet location '%s' AS SELECT * FROM %s",
+        () -> Actions.migrate(source).as(dest)
+        .withAdditionalProperty("dogs", "sundance")
+        .withAdditionalProperties(props),
+        2);
+
+    SparkTable table = loadTable(dest);
+
+
+    Map<String, String> expectedProps = Maps.newHashMap();
+    expectedProps.putAll(props);
+    expectedProps.put("dogs", "sundance");
+
+    for (Map.Entry<String, String> entry : expectedProps.entrySet()) {
+      Assert.assertTrue(
+          "Created table missing property " + entry.getKey(),
+          table.properties().containsKey(entry.getKey()));
+      Assert.assertEquals("Property value is not the expected value",
+          entry.getValue(), table.properties().get(entry.getKey()));
+    }
+  }
+
+  private SparkTable loadTable(String name) throws NoSuchTableException {
+    return (SparkTable) catalog.loadTable(Spark3Util.catalogAndIdentifier(spark, name).identifier());
+  }
+
+  // Creates a table, runs the migration command and checks the results.
+  private void testCreate(String source, String dest, String sqlCreate, Supplier<CreateAction> action,
+      long expectedMigratedFiles) throws
+      Exception {
+
+    File location = temp.newFolder();
+
+    spark.sql(String.format(sqlCreate, source, location, baseTableName));
+
+    long migratedFiles = action.get().execute();
+
+    SparkTable table = loadTable(dest);
+
+    Assert.assertEquals("Provider should be iceberg", "iceberg",
+        table.properties().get(TableCatalog.PROP_PROVIDER));
+    Assert.assertEquals("Expected number of migrated files", expectedMigratedFiles, migratedFiles);
+    Assert.assertEquals("Expected rows in table ", 3, spark.table(dest).count());
+  }
+
+  // Inserts records into the destination, makes sure those records exist and source table is unchanged
+  private void testIsolatedSnapshot(String source, String dest) {
+    List<Row> expected = spark.sql(String.format("SELECT * FROM %s", source)).collectAsList();
+
+    List<SimpleRecord> extraData = Lists.newArrayList(
+        new SimpleRecord(4, "d")
+    );
+    Dataset<Row> df = spark.createDataFrame(extraData, SimpleRecord.class);
+    df.write().format("iceberg").mode("append").saveAsTable(dest);
+
+    List<Row> result = spark.sql(String.format("SELECT * FROM %s", source)).collectAsList();
+    Assert.assertEquals("No additional rows should be added to the original table", expected.size(),
+        result.size());
+
+    List<Row> snapshot = spark.sql(String.format("SELECT * FROM %s WHERE id = 4 AND data = 'd'", dest)).collectAsList();
+    Assert.assertEquals("Added row not found in snapshot", 1, snapshot.size());
+  }
+
+  private String uniqueName(String source) {
```

Comment:
Why is this needed?

Reply:
This is mostly because the tests take way too long when I was running them locally because of HiveMetastore retries. When every test had a unique name I didn't have to drop tables and I wouldn't have to wait for the metastore to check for 2 minutes to see whether the table existed when I called DROP IF EXISTS. 

This is one of the reasons I brought up the retry length a while back, I just couldn't efficiently test this code on my local machine.

Reply:
I don't think a `DROP IF EXISTS` should take 2 minutes with the retry. That sounds like something in the catalog is broken. The retry should only happen if the metadata file is known, but can't be loaded. In that case, the table does exist.

Maybe the test before/after methods are dropping the temporary files before dropping the table?